git-secrets

```diff
@@ -338,9 +338,17 @@ case "${COMMAND}" in
   --scan-history) scan_with_fn_or_die "scan_history" "$@" ;;
   --list)
     if [ ${GLOBAL} -eq 1 ]; then
-      git config --global --get-regex secrets.*
+      RESULT=1
+      if git config --global --get-regex --type path "^secrets\.providers$"; then RESULT=0; fi
+      if git config --global --get-regex "^secrets\.patterns$"; then RESULT=0; fi
+      if git config --global --get-regex "^secrets\.allowed$"; then RESULT=0; fi
+      [ $RESULT -eq 0 ]
```

Comment:
The old regex was overly broad and should have been at least `^secrets\..*` to begin with.
`--type path` should only be applied to providers, so they have to be broken out here. That means explicitly enumerating the settings keys. Because the tests (and plausible real world use cases) expect `--list`'s exit code to reflect whether any settings were found I had to accumulate a result along the way, and this was the most concise way that I was comfortable with.